### ISSUE TITLE
Adopting new NumPy random generator

### ISSUE THREAD

**Issue by qiyunzhu** [scikit-bio/scikit-bio#1886]
NumPy adopted a new random number generator policy ([NEP 19](https://numpy.org/neps/nep-0019-rng-policy.html)) (see a [tutorial](https://albertcthomas.github.io/good-practices-random-number-generators/)) since version 1.17:

```python
rng = np.random.default_rng(42)
res = rng.integers(10000, size=10)
```

It is recommended over the convensional way:

```python
np.random.seed(42)
res = np.random.randint(10000, size=10)
```

**Consideration**: In addition to the rationales detailed in [NEP 19](https://numpy.org/neps/nep-0019-rng-policy.html), the new generator also has significant performance advantage in applications we care. Specifically, the rarefaction (i.e., subsampling) process, a very slow step in previous workflows, could gain 10x performance (see #1885 for details).

**Proposal**

Refactor scikit-bio codebase to replace the convensional `np.random` usages with the new random generator.

For example, a function previously written as:

```python
def subsample(counts, n):
    probs = counts / counts.sum()
    return np.random.multinomial(n, probs)
```

Should become:

```python
def subsample(counts, n, seed=None):
    rng = get_rand_gen(seed)
    probs = counts / counts.sum()
    return rng.multinomial(n, probs)
```

The global utility function `get_rand_gen` should be like (Here is a [similar one](https://github.com/scikit-learn/scikit-learn/blob/093e0cf14aff026cca6097e8c42f83b735d26358/sklearn/utils/validation.py#L1249) in scikit-learn, although it hasn't adopted the new random generator):

```python
def get_rand_gen(seed):
    if seed is None or isinstance(seed, int):
        return np.random.default_rng(seed)
    if isinstance(seed, np.random.Generator):
        return seed
    raise ValueError('Invalid seed.')
```

**Workload**: There are a total of 21 instances of `np.random` in the scikit-bio codebase (excluding unit and doc tests).

**Version control**: Scikit-bio currently requires NumPy >= 1.9.2. This needs to change to 1.17 in order to adopt the new random generator. Note that NumPy 1.17 was released in July 2019, which was before the release of Python 3.8 (Oct 2019). Since scikit-bio's Python version requirement is >= 3.8, it should be reasonable to require NumPy >= 1.17.

**Backward compatibility**: If support for NumPy < 1.17 is desired, it might be possible to use NumPy's [legacy random generation](https://numpy.org/doc/stable/reference/random/legacy.html) if the new random generator is not available.

```python
rs = np.random.RandomState(42)
res = rs.randint(10000, size=10)
```

It is possible to run code to check this when skbio is imported, and store it in a global configuration.

Note however, that some methods of the old `RandomState` and the new random generator are not identical. For example `randint` vs `integers`. Therefore, one needs to carefully consider the methods and parameters used in the specific skbio functions.

**Comment by mortonjt** [comment]
Hi, see this issue : https://github.com/scikit-bio/scikit-bio/issues/946

**Comment by qiyunzhu** [comment]
@mortonjt Cool! Given there has been consensus, let's make this a priority. Any concern on it? @wasade @gregcaporaso @ebolyen 

How about the NumPy version thing? My suggestion is to ditch support for NumPy < 1.17.
